src/reactive-object.ts

```diff
@@ -524,7 +524,7 @@ export class ReactiveObject {
         }
         iterateProperties(map ? args.slice(0, args.length - 1) : args);
         var observableList = finalProperties.map(prop => {
-            return this.whenSingle(prop, true);
+            return this.whenSingle(prop, true).distinctUntilChanged((x, y) => x.newPropertyValue === y.newPropertyValue);
```

Comment:
@sickboy Probably the best place to put the distinct() call is to put it in [`ReactiveObject.whenSingle()`](https://github.com/KallynGowdy/RxUI/blob/master/src/reactive-object.ts#L341). That way the improvement can be shared to more implementations than just `#whenAny()`.

There are two returns from the [`ReactiveObject.whenSingle()`](https://github.com/KallynGowdy/RxUI/blob/master/src/reactive-object.ts#L341) method, so you probably will want to distinct both of them. It would be nice to refactor it so that less work is happening, but that is probably not going to happen anytime soon.

Thanks for the support!